### PR TITLE
removed old way of handling addons and created another way to add addons

### PR DESCRIPTION
**Pull request by ypwalter**
@ShakoHo @askeing r?

### CODE REVIEW

File: settings/default.json [hunk 1]
```diff
@@ -27,5 +27,32 @@
     "folder": "0BwkEhia_D6l_bVJ0aDdGSU5pMzg",
     "filename": "cookies_gsuite_fb.sqlite"
   },
-  "extensions": {}
+  "extensions": {
```
Comment (askeing) [inline settings/default.json]: please remove this object, the key is duplicate.

File: settings/default.json [hunk 2]
```diff
@@ -27,5 +27,32 @@
...
+    "Gecko Profiler 2.0.18": ["gecko_profiler", true]
+  },
+  "extensions": {
+    "GeckoProfiler": {
+      "version": "Gecko Profiler 2.0.18",
+      "enable": true,
```
Comment (askeing) [inline settings/default.json]: the default settings shouldn't enable `gecko profiler`.

File: settings/default.json (hunk 1 above)
Comment (ypwalter) [inline settings/default.json]: done, thanks.

**Comment by ypwalter** [comment]
@askeing @ShakoHo r?

**Comment by ypwalter** [comment]
Don't merge before lunar new year vacation, but please review it when you come back.

**Comment by ShakoHo** [comment]
lgtm r+

**Comment by askeing** [comment]
Ref: #493

**Pull request merged**